### ISSUE TITLE
`TypedDict` silently accepts experimental PEP 728 arguments but ignores them

### ISSUE THREAD

**Issue by decorator-factory** [microsoft/pyright#9928]
For completely unknown `TypedDict` kwargs, pyright correctly shows an error:

```py
class Foo(TypedDict, bar=True):
    #                ^^^^^^^^
    # TypedDict does not support __init_subclass__ parameter "bar
    x: int
```

However, it does not complain about `closed` and `extra_items`, even if `enableExperimentalFeatures` is not enabled.

```py
from typing import TypedDict

class HasX(TypedDict, closed=True):  # no error reported
    x: int

class HasXY(TypedDict):
    x: int
    y: int

has_xy: HasXY = {"x": 1, "y": 2}
has_x: HasX = has_xy  # assignment is allowed
```

(the assignment is disallowed when `enableExperimentalFeatures` is on)

It makes it seem like these flags are supported, but they are silently ignored. I expected an error message like  

> PEP 728 support is experimental. Set `enableExperimentalFeatures` to `true` to enable it

---

Pyright version: 1.1.394

Python version: 3.12

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.395.
